examples/software-device-advanced/active_obj.cpp

```diff
@@ -0,0 +1,309 @@
+            }
+            if (profiles != active_profiles) {
+                // Need to update streams.
+                if (legacy_dev->is_streaming()) legacy_dev->stop();
+
+                // Disable all old streams
+                for (int i = 0; i<int(rs::stream::points); ++i) {
+                    auto stream = rs::stream(i);
+                    if (legacy_dev->is_stream_enabled(stream))
+                        legacy_dev->disable_stream(stream);
+                }
+
+                // Enable all active streams
+                for (auto& kvp : profiles) {
+                    auto vp = kvp.second;
+                    legacy_dev->enable_stream(stream_iconv(vp.stream_type()), vp.width(),
+                        vp.height(), fmt_iconv(vp.format()), vp.fps());
+                    legacy_dev->set_frame_callback(stream_iconv(vp.stream_type()), [this](rs::frame f) { 
+                        on_frame(std::move(f)); 
+                    });
+                }
+                active_profiles.swap(profiles);
+                legacy_dev->start();
+            }
+
+            // Poll changes to options
+            for (auto & kvp : cur_options) {
+                rs2_stream stream;
+                rs2_option opt;
+                bool is_writable;
+                std::tie(stream, opt, is_writable) = option_conv(kvp.first);
+                auto sensor = sensors.at(stream);
+                if (is_writable) {
+                    auto new_val = sensor.get_option(opt);
+                    if (new_val != kvp.second) {
+                        kvp.second = new_val;
+                        if (opt == RS2_OPTION_DEPTH_UNITS) new_val *= 1e6; // Legacy API uses micrometers, Modern API uses meters.
+                        legacy_dev->set_option(kvp.first, new_val);
+
+                        // Setting these options causes the auto control to change.
+                        if (kvp.first == rs::option::color_exposure) {
+                            cur_options[rs::option::color_enable_auto_exposure] = 0;
+                            sensor.set_option(RS2_OPTION_ENABLE_AUTO_EXPOSURE, 0);
+                        }
+                        else if (kvp.first == rs::option::color_white_balance) {
+                            cur_options[rs::option::color_enable_auto_white_balance] = 0;
+                            sensor.set_option(RS2_OPTION_ENABLE_AUTO_EXPOSURE, 0);
+                        }
+                        else if (kvp.first == rs::option::r200_lr_exposure) {
+                            cur_options[rs::option::r200_lr_auto_exposure_enabled] = 0;
+                            sensor.set_option(RS2_OPTION_ENABLE_AUTO_EXPOSURE, 0);
+                        }
+                        else if (kvp.first == rs::option::fisheye_exposure) {
+                            cur_options[rs::option::fisheye_color_auto_exposure] = 0;
+                            sensor.set_option(RS2_OPTION_ENABLE_AUTO_EXPOSURE, 0);
+                        }
```

Comment:
Final (default) `else` is missing to cover all the unresolved cases. Probably would also generate a log

Reply:
There is no final else because these are the unique cases. All options go through Line 88 where the core logic happens, this if/else set (I could replace it with a switch) is just to handle the unique cases where manually adjusting expose/white balance turns off automatic management of these values